### PR TITLE
Removed unused variables

### PR DESCRIPTION
**Pull request by jonathanp**
I've gone through the components and removed all unused variables I could find.

### CODE REVIEW

File: fields/mixins/ArrayField.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-var React = require('react');
```
Comment (JedWatson) [inline fields/mixins/ArrayField.js]: React will get used when the JSX is transpired... there are a few of these.

**Comment by JedWatson** [comment]
Awesome, thank you! I wonder why our linter isn't complaining about these. I've noticed it missed quite a few things and suspect our config may not be correct (maybe related to an ESLint update...?)

I think there are a couple of errors in here because `React` is used when JSX is transpiled, but aside from that this looks great!

**Comment by jonathanp** [comment]
Good catch! I've re-added those `var React = require('react')`'s now, except in `admin/client/columns.js` where there's no JSX.

**Comment by JedWatson** [comment]
:+1:

**Pull request merged**